src/python_testing/TC_ECOINFO_2_1.py

```diff
@@ -15,20 +15,92 @@
 #    limitations under the License.
 #
 
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: examples/fabric-admin/scripts/fabric-sync-app.py
+#     app-args: --app-admin=${FABRIC_ADMIN_APP} --app-bridge=${FABRIC_BRIDGE_APP} --stdin-pipe=dut-fsa-stdin --discriminator=1234
+#     script-args: >
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --string-arg th_server_app_path:${ALL_CLUSTERS_APP} dut_fsa_stdin_pipe:dut-fsa-stdin
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     script-start-delay: 5
+#     factoryreset: true
+#     quiet: false
+# === END CI TEST ARGUMENTS ===
+
+import asyncio
+import logging
+import random
+import tempfile
+
 import chip.clusters as Clusters
 from chip.clusters.Types import NullValue
 from chip.interaction_model import Status
 from chip.tlv import uint
 from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main, type_matches
 from mobly import asserts
+from TC_MCORE_FS_1_1 import AppServer
 
 
 class TC_ECOINFO_2_1(MatterBaseTest):
 
+    @async_test_body
+    async def setup_class(self):
+        super().setup_class()
+
+        self.th_server = None
+        self.storage = None
+
+        # Get the path to the TH_SERVER app from the user params.
+        if th_server_app := self.user_params.get("th_server_app_path"):
+            # Create a temporary storage directory for keeping KVS files.
+            self.storage = tempfile.TemporaryDirectory(prefix=self.__class__.__name__)
+            logging.info("Temporary storage directory: %s", self.storage.name)
+
+            # Get the named pipe path for the DUT_FSA app input from the user params.
+            if dut_fsa_stdin_pipe := self.user_params.get("dut_fsa_stdin_pipe"):
+                self.dut_fsa_stdin = open(dut_fsa_stdin_pipe, "w")
+
+            self.th_server_port = 5544
+            self.th_server_discriminator = random.randint(0, 4095)
+            self.th_server_passcode = 20202021
+
+            # Start the server app.
+            self.th_server = AppServer(
+                th_server_app,
+                storage_dir=self.storage.name,
+                port=self.th_server_port,
+                discriminator=self.th_server_discriminator,
+                passcode=self.th_server_passcode)
+            self.th_server.start()
+
+            # Add some server to the DUT_FSA's Aggregator/Bridge.
+            self.dut_fsa_stdin.write(
```

Comment:
`self.dut_fsa_stdin` doesn't exist if dut_fsa_stdin_pipe is False

Reply:
Fixed